spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.spec.js

```diff
@@ -0,0 +1,295 @@
+/* eslint-env jasmine */
+
+describe('Google Analytics ecommerce tracking', function () {
+  var GOVUK = window.GOVUK
+  var searchResultsParentEl
+  var searchResults
+  var resultsMetaTag
+  var resultToBeClicked
+  var onPageLoadExpected
+  var onSearchResultClickExpected
+  var pageViewExpected
+
+  function agreeToCookies () {
+    GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+  }
+
+  beforeEach(function () {
+    resultsMetaTag = document.createElement('meta')
+    resultsMetaTag.setAttribute('name', 'govuk:search-result-count')
+    resultsMetaTag.setAttribute('content', '12345')
+
+    searchResultsParentEl = document.createElement('div')
+    searchResultsParentEl.setAttribute('data-search-query', 'test-search-query')
+    searchResultsParentEl.setAttribute('data-module', 'ga4-enhanced-ecommerce-tracker')
+    searchResultsParentEl.setAttribute('data-ecommerce-variant', 'test-ecommerce-variant')
+    searchResultsParentEl.setAttribute('data-list-title', 'test-list-title')
+    searchResultsParentEl.setAttribute('data-ecommerce-start-index', '1')
+
+    searchResults = document.createElement('div')
+    searchResults.innerHTML =
+      '<a data-ecommerce-path="/coronavirus" data-ecommerce-row="1" data-ecommerce-index="1" data-track-category="navFinderLinkClicked" data-track-action="Search.1" data-track-label="/coronavirus" data-track-options="{&quot;dimension28&quot;:20,&quot;dimension29&quot;:&quot;Coronavirus (COVID-19): guidance and support&quot;}" class="gem-c-document-list__item-title   govuk-link" href="/coronavirus">Coronavirus (COVID-19): guidance and support</a>' +
```

Comment:
Might be worth removing the data values and the css class which aren't needed for these elements, so it's easier to read

Reply:
Agreed, I've removed those.